### DISCUSSION TITLE
Add max_title_length to TabBarData

### DISCUSSION THREAD

**Discussion by alaricljs** [kovidgoyal/kitty#5330] [Ideas]
I'd like to see `max_title_length` added to `TabBarData` and `draw_title().eval_locals` so that `tab_title_template`'s can contain formatting like: `{title:<{max_title_length}}` which would achieve full width tab options with no user implemented code.

1 tab:
![2-region](https://user-images.githubusercontent.com/5070475/181627488-ea2acb71-4363-4cf1-90a4-b182773607b6.jpg)
2 tabs (right active):
![3-region](https://user-images.githubusercontent.com/5070475/181627637-004b12c0-d8af-4009-8396-b2f12fe87db9.jpg)
3 tabs (middle active)
![4-region](https://user-images.githubusercontent.com/5070475/181627719-e715d0e9-f6f6-4823-8f34-2237b8b85077.jpg)

```
tab_bar_edge top
tab_bar_margin_width 2.0
tab_bar_margin_height 5.0 2.0
tab_bar_style separator
tab_bar_align center
tab_bar_min_tabs 1
tab_separator ""
tab_title_template "{fmt.fg.red}{bell_symbol}{activity_symbol}{fmt.fg.tab} {title:<{max_title_length-2}} "
active_tab_title_template "\x1b[4:1m{fmt.fg.red}{bell_symbol}{activity_symbol}{fmt.fg.tab} {title:<{max_title_length-2}} \x1b[4:0m"
active_tab_foreground   #268bd2
active_tab_background   #002b36
active_tab_font_style   bold
inactive_tab_foreground #fdf6e3
inactive_tab_background #073642
inactive_tab_font_style normal
```

Yes, I know - do it myself and send a PR.  Tonight I head to the hospital for unknown days, so maybe someone else does it before I get back?

**Comment by kovidgoyal** [comment]
https://github.com/kovidgoyal/kitty/commit/0ce2a23af30afa6a3e74f92217f0cee502dfc475

**Comment by kovidgoyal** [comment]
https://github.com/kovidgoyal/kitty/commit/0ce2a23af30afa6a3e74f92217f0cee502dfc475
